### DISCUSSION TITLE
Donate SpecFlow.DependencyInjection plugin to Reqnroll?

### DISCUSSION THREAD

**Discussion by mbhoek** [reqnroll/Reqnroll#19] [Contributor Q&A] [answered]
Hi @gasparnagy, I'm the maintainer of the [SpecFlow.DependencyInjection](https://github.com/solidtoken/SpecFlow.DependencyInjection) plugin which enables to use `Microsoft.Extensions.DependencyInjection` for resolving test dependencies.

Congratulations on starting Reqnroll. I am happy to migrate my plugin to Reqnroll, but I am also willing to donate it to the Reqnroll codebase (i.e. make it part of [`Reqnroll/Plugins`](https://github.com/reqnroll/Reqnroll/tree/main/Plugins)). Is this something you'd be interested in, or would you rather have me keep it as a seperate plugin?

Thanks for your time,
Mark

**Comment by gasparnagy** [comment, marked as answer]
@mbhoek Thank you for you ongoing work on the plugin.

We would be honoed to have it in the main repo.

Please try to make a PR with the repo included and let me know if you run into any thoubles with it.

**Reply by mbhoek** [reply]
Good to hear. I could use a little help with deciding on a name for the plugin. Does `Reqnroll.DependencyInjection.ReqnrollPlugin` work for you?

**Reply by gasparnagy** [reply]
The NuGet package of the similar Autofac and Windsor plugins are called [Reqnroll.Autofac](https://www.nuget.org/packages/Reqnroll.Autofac) amd [Reqnroll.Windsor](https://www.nuget.org/packages/Reqnroll.Windsor), but due to the way how the plugins are discovered, the assemblies that implement the plugin inside the package are called `Reqnroll.Autofac.ReqnrollPlugin` and `Reqnroll.Windsor.ReqnrollPlugin`.

For `Microsoft.Extensions.DependencyInjection` it is a little harder as that does not have a short name, like Autofac. So we would need to pick the name how the community usually refers to it. Is that `Microsoft.Extensions.DependencyInjection`, or `Microsoft.DependencyInjection` or just `DependencyInjection`? I did a quick research and I mainly found it called `Microsoft.Extensions.DependencyInjection`, but maybe you know this better. But once we have the name, we can name the package and the plugin accordingly, e.g. `Reqnroll.Microsoft.Extensions.DependencyInjection`/`Reqnroll.Microsoft.Extensions.DependencyInjection.ReqnrollPlugin`.

For the project configuration & setup (because many project configurations are configured globally for the Reqnroll solution), please check one of the existing ones, e.g.: https://github.com/reqnroll/Reqnroll/tree/main/Plugins/Reqnroll.Autofac.ReqnrollPlugin

**Reply by mbhoek** [reply]
`Reqnroll.Microsoft.Extensions.DependencyInjection` is a *really* long name but it does make sense, so I think I'll use that. Thanks!

**Reply by MarioMajcicaAtABNAMRO** [reply]
Any updates on this plugin?

**Reply by mbhoek** [reply]
@MarioMajcicaAtABNAMRO Subscribe to [this issue](https://github.com/solidtoken/SpecFlow.DependencyInjection/issues/106) to get updates.

**Comment by MarianKrkosMews** [comment]
Hey guys, how's it going with plugin integration? This's the last thing I'm waiting for, before migrating to Reqnroll. There was no update for few weeks in the PR, so I would like to know if there is some problem, or it's just lack of time?

**Reply by gasparnagy** [reply]
Lack of time. I will try to help next week to have this (and the other things we need for making a new release) ready.

**Reply by mbhoek** [reply]
@gasparnagy Thanks. Please wait until Monday, I'm about to push new work this weekend.

**Reply by gasparnagy** [reply]
@mbhoek Sure, no prob. I can deal with it Tuesday or Wednesday the earliest.

**Comment by gasparnagy** [comment]
It is out: https://github.com/orgs/reqnroll/discussions/142

The package is here: https://www.nuget.org/packages/Reqnroll.Microsoft.Extensions.DependencyInjection

closing this topic
